File: python/credit_check.py
def credit_check(str):
    lis = list(str)
    ans = []
    sum = 0
    for j, i in enumerate(lis[::-1]):
        num = int(i)
        ele = 0
        if j % 2 == 1:
            ele = num*2
        else:
            ele = num

        if ele >=10:
            ele = ele%10 + (ele-(ele%10))//10
            ans.append(ele)
        else:
            ans.append(ele)

        sum = sum + ele
    
    #print(sum)
    return "The number is valid!" if sum%10==0 else "The number is invalid!"

str = '5541808923795240'
print(credit_check(str))
# Your Luhn Algorithm Here
# Expected Output:
# If it is valid, print "The number is valid!"
# If it is invalid, print "The number is invalid!"

File: js/creditCheck.js
exports.creditCheck = function(str){
    const arr = str.split('')
    let j = 0;
    let sum = 0
    for(let i = arr.length -1; i>=0; i--){
        if(j%2 === 1)
            arr[i]*=2
        j++

        if(arr[i]>=10)
            arr[i]=(arr[i]%10)+((arr[i]-(arr[i]%10))/10)

        sum+=parseInt(arr[i],10)
    }

    if(sum%10===0)
        return("The number is valid!")

    return("The number is invalid!")
}

//console.log(creditCheck('5541808923795240'))


